addons/account_voucher/views/account_voucher_views.xml

```diff
@@ -230,16 +230,21 @@
                         <group>
                             <field name="voucher_type" invisible="True"/>
                             <field name="currency_id" invisible="True"/>
-                            <field name="company_id" options="{'no_create': True}" groups="base.group_multi_company"/>
                             <field name="partner_id" domain="[('customer','=',True)]" string="Customer" context="{'search_default_customer':1, 'show_address': 1}" options='{"always_reload": True}'/>
                             <field name="pay_now" required="1"/>
-                            <field name="account_id"/>
+                            <field name="payment_journal_id"
+                                   attrs="{'invisible': [('pay_now', '!=', 'pay_now')], 'required': [('pay_now', '=', 'pay_now')]}"/>
+                            <field name="account_id" attrs="{'invisible': [('pay_now', '=', 'pay_now')]}"/>
+                            <field name="company_id" options="{'no_create': True}" groups="base.group_multi_company"/>
                         </group>
                         <group>
-                            <field name="journal_id" domain="[('type','=','sale')]" widget="selection" groups="account.group_account_user"/>
+                            <field name="journal_id"
+                                   domain="[('type','=','sale')]"
+                                   widget="selection"
+                                   groups="account.group_account_user"/>
                             <field name="date"/>
                             <field name="date_due" attrs="{'invisible':[('pay_now','=','pay_now')]}"/>
-                            <field name="name" attrs="{'invisible': [('pay_now', '=', 'pay_later')]}"/>
+                            <field name="name" string="Payment Memo" attrs="{'invisible': [('pay_now', '=', 'pay_later')]}"/>
```

Comment:
Alas, you can't change/add a label in stable unless you update the pot file in order to make it translatable. I'd simply skip that change for v10

Reply:
ok